src/executor/executor.cc

```diff
@@ -152,7 +153,8 @@ struct Executor::Impl {
         return this->contexts[key];
       }
       plan.impl_->operationsReset();
-      plan.impl_->lightLoadExecutionPlan(inputMessageSize, outputMessageSize, constSrcOffset, constDstOffset);
+      plan.impl_->lightLoadExecutionPlan(inputMessageSize, outputMessageSize, constSrcOffset, constDstOffset, rank,
```

Comment:
Seems we need to add flag % 2 to `DeviceExecutionPlanKey`, otherwise we will not recalculate the scratch buffer offset, even flag is changed